### PR TITLE
Port across devtools enhancements

### PR DESCRIPTION
**Pull request by janekptacijarabaci**
Ad #31 

This resolves partially. I will continue - as soon as there is time...

---

[moebius#54: DevTools: Inspector - tooltip get element width/height](https://github.com/MoonchildProductions/moebius/pull/54)

[moebius#55: DevTools: gcli commands - cookie; + fix typo: "gcli." vs. "l10n."](https://github.com/MoonchildProductions/moebius/pull/55)

__DevTools: The cmd line - "-help" and "-jsdebugger"__
[moebius#58](https://github.com/MoonchildProductions/moebius/pull/58)
[moebius#250](https://github.com/MoonchildProductions/moebius/pull/250)
[moebius#256](https://github.com/MoonchildProductions/moebius/pull/256)

[moebius#63: DevTools: Network - proxy - remoteAddress](https://github.com/MoonchildProductions/moebius/pull/63)

[moebius#95: DevTools: Inspector - data: URL source links](https://github.com/MoonchildProductions/moebius/pull/95)

[moebius#50: DevTools: Scratchpad - Reload and Run](https://github.com/MoonchildProductions/moebius/pull/50)
\- `Shift-Alt-R` (`Ctrl-Alt-R` does not work now)

[moebius#93: DevTools: Network - transferred size](https://github.com/MoonchildProductions/moebius/pull/93)
\- Required: [bug 1309194](https://bugzilla.mozilla.org/show_bug.cgi?id=1309194), [bug 1317205](https://bugzilla.mozilla.org/show_bug.cgi?id=1317205)

\- Not implemented: `DOMContentLoaded` and `load` ([bug 1309866](https://bugzilla.mozilla.org/show_bug.cgi?id=1309866))

[moebius#112: DevTools: Browser Console - restore sessions](https://github.com/MoonchildProductions/moebius/pull/112)

__DevTools: Network - security (improvements)__
[moebius#113](https://github.com/MoonchildProductions/moebius/pull/113)
[moebius#118](https://github.com/MoonchildProductions/moebius/pull/118)
[moebius#127](https://github.com/MoonchildProductions/moebius/pull/127)

[moebius#116: DevTools: Network - the secureConnectionStart/the PerformanceTiming](https://github.com/MoonchildProductions/moebius/pull/116)

[moebius#124: DevTools: Style editor - StyleSheetsActor](https://github.com/MoonchildProductions/moebius/pull/124)

[moebius#166: DevTools: gcli commands - screenshot - the vertical mode](https://github.com/MoonchildProductions/moebius/pull/166)

[moebius#167; DevTools: Network - Copy as cURL (POST)](https://github.com/MoonchildProductions/moebius/pull/167)

[moebius#257: DevTool: The cmd line - -browserconsole](https://github.com/MoonchildProductions/moebius/pull/257)

[moebius#290: DevTools: Scratchpad - source-editor commands controller](https://github.com/MoonchildProductions/moebius/pull/290)

[moebius#302: DevTools: Actors - tab - NS_BINDING_ABORTED](https://github.com/MoonchildProductions/moebius/pull/302)

[moebius#348: DevTools: gcli commands - "help cmd" and "cmd refresh"](https://github.com/MoonchildProductions/moebius/pull/348)

[moebius#162: DevTools: Throws an error - String contains an invalid character](https://github.com/MoonchildProductions/moebius/pull/162)

[moebius#323: DevTools: Network - raw headers - the Status Line](https://github.com/MoonchildProductions/moebius/pull/323)

[moebius#352: DevTools: Network - Open in New Tab](https://github.com/MoonchildProductions/moebius/pull/352)

### CODE REVIEW

**Comment by mattatobin** [comment]
This will leave DevTools functional correct?

File: devtools/shared/webconsole/network-monitor.js [hunk 1]
```diff
@@ -1327,8 +1329,24 @@ NetworkMonitor.prototype = {
 
     let response = {};
     response.httpVersion = statusLineArray.shift();
-    response.remoteAddress = httpActivity.channel.remoteAddress;
-    response.remotePort = httpActivity.channel.remotePort;
+    // XXX: 
+    // Sometimes, when using a proxy server (manual proxy configuration),
+    // throws an errors:
```
Comment (wolfbeast) [inline devtools/shared/webconsole/network-monitor.js]: nit: "throws an error"

File: dom/performance/PerformanceTiming.cpp [hunk 2]
```diff
@@ -273,6 +301,22 @@ PerformanceTiming::ConnectStart()
...
 }
 
+DOMHighResTimeStamp
+PerformanceTiming::SecureConnectionStartHighRes()
+{
+  if (!nsContentUtils::IsPerformanceTimingEnabled() || !IsInitialized()) {
+    return mZeroTime;
+  }
+  return mSecureConnectionStart.IsNull() ? mZeroTime
+                                         : TimeStampToDOMHighRes(mSecureConnectionStart);
```
Comment (wolfbeast) [inline dom/performance/PerformanceTiming.cpp]: Make sure to clamp the HighRes performance timers, wherever used, to a resolution of 1 ms!

Comment (janekptacijarabaci) [reply dom/performance/PerformanceTiming.cpp]: If you prefer another solution, you can rewrite it according to how you conceive it to be.

**Pull request merged**